src/mame/drivers/mc10.cpp

```diff
@@ -24,125 +24,116 @@
-		: driver_device(mconfig, type, tag)
-		, m_maincpu(*this, "maincpu")
-		, m_mc6847(*this, "mc6847")
-		, m_ef9345(*this, "ef9345")
-		, m_dac(*this, "dac")
-		, m_ram(*this, RAM_TAG)
-		, m_cassette(*this, "cassette")
-		, m_printer(*this, "printer")
-	{ }
-
-	required_device<m6803_cpu_device> m_maincpu;
-	optional_device<mc6847_base_device> m_mc6847;
-	optional_device<ef9345_device> m_ef9345;
-	required_device<dac_bit_interface> m_dac;
-	required_device<ram_device> m_ram;
-	required_device<cassette_image_device> m_cassette;
-	required_device<printer_image_device> m_printer;
-
-	uint8_t *m_ram_base;
-	uint32_t m_ram_size;
-	uint8_t m_keyboard_strobe;
-	uint8_t m_port2;
-
-	// printer
-	uint8_t m_pr_buffer;
-	uint8_t m_pr_counter;
-	uint8_t m_pr_state;
-
-	DECLARE_READ8_MEMBER( mc10_bfff_r );
-	DECLARE_WRITE8_MEMBER( mc10_bfff_w );
-	DECLARE_READ8_MEMBER( alice90_bfff_r );
-	DECLARE_WRITE8_MEMBER( alice32_bfff_w );
-	DECLARE_READ8_MEMBER( mc10_port1_r );
-	DECLARE_WRITE8_MEMBER( mc10_port1_w );
-	DECLARE_READ8_MEMBER( mc10_port2_r );
-	DECLARE_WRITE8_MEMBER( mc10_port2_w );
-	DECLARE_READ8_MEMBER( mc6847_videoram_r );
-	DECLARE_DRIVER_INIT(mc10);
-	TIMER_DEVICE_CALLBACK_MEMBER(alice32_scanline);
+	class mc10_state : public driver_device
+	{
+	public:
+		mc10_state(const machine_config &mconfig, device_type type, const char *tag);
+
+		DECLARE_READ8_MEMBER(mc10_bfff_r);
+		DECLARE_WRITE8_MEMBER(mc10_bfff_w);
+		DECLARE_READ8_MEMBER(mc10_port1_r);
+		DECLARE_WRITE8_MEMBER(mc10_port1_w);
+		DECLARE_READ8_MEMBER(mc10_port2_r);
+		DECLARE_WRITE8_MEMBER(mc10_port2_w);
+		DECLARE_READ8_MEMBER(alice90_bfff_r);
+		DECLARE_WRITE8_MEMBER(alice32_bfff_w);
+
+		DECLARE_READ8_MEMBER(mc6847_videoram_r);
+		TIMER_DEVICE_CALLBACK_MEMBER(alice32_scanline);
+
+	protected:
+		// device-level overrides
+		virtual void device_start() override;
+
+	private:
+		//printer state
+		enum class printer_state
+		{
+			WAIT,
+			REC,
+			DONE
+		};
+
+		required_device<m6803_cpu_device> m_maincpu;
+		optional_device<mc6847_base_device> m_mc6847;
+		optional_device<ef9345_device> m_ef9345;
+		required_device<dac_bit_interface> m_dac;
+		required_device<ram_device> m_ram;
+		required_device<cassette_image_device> m_cassette;
+		required_device<printer_image_device> m_printer;
+		required_ioport_array<8> m_pb;
+		required_memory_bank m_bank1;
+		required_memory_bank m_bank2;
+
+		uint8_t *m_ram_base;
+		uint32_t m_ram_size;
+		uint8_t m_keyboard_strobe;
+		uint8_t m_port2;
+
+		// printer
+		uint8_t m_pr_buffer;
+		uint8_t m_pr_counter;
+		printer_state m_pr_state;
+
+		uint8_t read_keyboard_strobe(bool single_line);
+	};
 };
 
 
 /***************************************************************************
     MEMORY MAPPED I/O
 ***************************************************************************/
 
-READ8_MEMBER( mc10_state::mc10_bfff_r )
+uint8_t mc10_state::read_keyboard_strobe(bool single_line)
```

Comment:
Out-of-line implementations need to be in the namespace of their class.  This needs to be inside the anonymous namespace.

This is partly why I frown on indenting for namespaces - you end up having to indent the entire file.  It's better to just try to keep at least the bulk of a file in a single namespace, so you know it's going to be opened near the top and closed near the bottom.